tools/stonith_admin.c

```diff
@@ -323,57 +325,31 @@ handle_level(stonith_t *st, char *target, int fence_level,
                                        name, value, fence_level);
 }
 
-static char *
-fence_action_str(const char *action)
-{
-    char *str = NULL;
-
-    if (action == NULL) {
-        str = strdup("unknown");
-    } else if (action[0] == 'o') { // on, off
-        str = crm_concat("turn", action, ' ');
-    } else {
-        str = strdup(action);
-    }
-    return str;
-}
-
-static void
-print_fence_event(stonith_history_t *event)
-{
-    char *action_s = fence_action_str(event->action);
-    time_t complete = event->completed;
-
-    printf("%s was able to %s node %s on behalf of %s from %s at %s\n",
-           (event->delegate? event->delegate : "This node"), action_s,
-           event->target, event->client, event->origin, ctime(&complete));
-    free(action_s);
-}
-
 static int
 handle_history(stonith_t *st, const char *target, int timeout, int quiet,
-             int verbose, int cleanup, int broadcast)
+             int verbose, int cleanup, int broadcast, pcmk__output_t *out)
 {
     stonith_history_t *history = NULL, *hp, *latest = NULL;
     int rc = 0;
 
     if (!quiet) {
         if (cleanup) {
-            printf("cleaning up fencing-history%s%s\n",
-                   target?" for node ":"", target?target:"");
+            out->info(out, "cleaning up fencing-history%s%s\n",
+                      target ? " for node " : "", target ? target : "");
```

Comment:
I think it looks odd that \n is passed with out->info() and no other functions. I'd rather it be called without the \n, and have the text implementation tack it on.